diff --git a/tb/ariane_tb.cpp b/tb/ariane_tb.cpp
index c64c4db..8b25e15 100644
--- a/tb/ariane_tb.cpp
+++ b/tb/ariane_tb.cpp
@@ -100,12 +100,12 @@ EMULATOR DEBUG OPTIONS (only supported in debug build -- try `make debug`)\n",
          );
 }
 
-int main(int argc, char **argv) {
+extern "C" int ariane_main(int argc, char **argv) {
   std::clock_t c_start = std::clock();
   auto t_start = std::chrono::high_resolution_clock::now();
   bool verbose;
   bool perf;
-  unsigned random_seed = (unsigned)time(NULL) ^ (unsigned)getpid();
+  unsigned random_seed = 0;
   uint64_t max_cycles = -1;
   int ret = 0;
   bool print_cycles = false;
